### PR TITLE
fix: 🐛 Replace `apollo-client` with `@apollo/client`

### PR DESCRIPTION
**Pull request by prashantasdeveloper**
### Description

This fixes the compilation error with typing middleware object
```
Type 'ApolloClient<NormalizedCacheObject>' is missing the following properties from type 'ApolloClient<NormalizedCacheObject>': refetchQueries, getObservableQueries, setLink
 ```

### Breaking Changes

NA

### JIRA Link

DA-769

### Checklist

- [ ] Updated the Readme.md (if required) ?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -94,6 +94,7 @@
     "prettier": "2.5.1",
     "prettier-eslint": "13.0.0",
     "prettier-eslint-cli": "5.0.1",
+    "react": "^18.2.0",
```
[the remark this answers is not in the document]
Comment (prashantasdeveloper) [reply package.json:97]: Yes 'react' is needed as a dep with @apollo/client

File: src/middleware/queries.ts [hunk 2]
```diff
@@ -29,7 +29,7 @@ import { GraphqlQuery } from '~/types/internal';
  */
 export function proposalVotes(
   variables: QueryProposalVotesArgs
-): GraphqlQuery<QueryProposalVotesArgs> {
+): QueryOptions<QueryProposalVotesArgs> {
```
[the remark this answers is not in the document]
Comment (prashantasdeveloper) [reply src/middleware/queries.ts:32]: It shouldn't be in my opinion. Also, `GraphqlQuery` was defined for internal purposes I assume so if anyone's if they are using it to strongly type something, shouldn't. Feels safe to mark it as a normal change

**Review by VictorVicente — APPROVED** [review]

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (polymath-eric) [reply package.json:97]: At least its dev only, but weird. Kinda sucks `.json` doesn't have comments for cases like these

[the remark this answers is not in the document]
Comment (polymath-eric) [reply package.json:97]: IDK what's wrong, but this seems like a bug or is unintened. Maybe something to do with webpack config or node modules given the `.cjs` part of the error.

This comment makes me think its supposed to be fixed even though we are using the newest versions:
https://github.com/apollographql/apollo-client/issues/6532#issuecomment-1163230666

**Pull request merged**